experimental/streaming/sdk/metric.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sdk
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/api/core"
+	"go.opentelemetry.io/api/metric"
+	"go.opentelemetry.io/experimental/streaming/exporter/observer"
+)
+
+type metricHandle struct {
+	instrument metric.Instrument
+	scope      observer.ScopeID
+}
+
+func (h *metricHandle) RecordAt(ctx context.Context, t time.Time, value float64) {
```

Comment:
`t time.Time` parameter seems to be ignored. Should we set the `Time` field of the `Event` with it?

Reply:
Actually, what's the meaning of the `time.Time` parameter? I see it can be set for each record when recording metrics one by one, but it can't be set for the batch updates at all. Should batch updates have a timestamp for each measurement or one for all of them or none at all (as it is currently)?

Reply:
I'm removing it. It's something we could specify later. I'm not sure how important users will find the ability to provide custom timestamps on metrics. We have custom timestamps on spans, but it's less clear if this is needed for metrics.

Reply:
> I'm not sure how important users will find the ability to provide custom timestamps on metrics.

The primary motivation is that `time` is a critical part of time series data, so the core abstract should contain it even if the layers above do not.  In the past I've used the zero value of `time.Time` to imply "real time".  It could be possible people have their own buffering above open telemetry and want to use the timestamp of values when they enter that, not when they enter open telemetry.